src/sage/graphs/graph_decompositions/slice_decomposition.pyx

```diff
@@ -0,0 +1,970 @@
+            sage: SD.lexicographic_label('f')
+            ['x', 'a', 'c', 'd']
+            sage: pos = lambda v: SD.lexBFS_order().index(v)
+            sage: set(SD.lexicographic_label('f')) \
+            ....:   == {v for v in G.neighbors('f') if pos(v) < pos('f')}
+            True
+
+        TESTS:
+
+            sage: SD = graphs.PetersenGraph().slice_decomposition()
+            sage: SD.lexicographic_label('Eric')
+            Traceback (most recent call last):
+            ...
+            LookupError: vertex (Eric) does not appear in the slice decomposition
+        """
+        if v not in self.sigma_inv:
+            raise LookupError(f"vertex ({v}) does not appear in the slice "
+                              "decomposition")
+        i = self.sigma_inv[v]
+        return self._xslice_lex_label(i)
+
+    def active_edges(self, v):
+        r"""
+        Return the active edges of the vertex `v`.
+
+        An edge `(u, w)` is said to be active for `v` if `u` and `w` belongs
+        to two differents slices of the x-slice sequence of `v`. Note that it
+        defines a partition of the edges of the underlying graph.
+
+        INPUT:
+
+        - ``v`` -- a vertex of the graph corresponding to the slice
+          decomposition.
+
+        OUTPUT:
+
+        A list of edges
+
+        EXAMPLES:
+
+        ::
+
+            sage: G = Graph('L~mpn~Nrv{^o~_').relabel('abcdefguvwxyz',inplace=False)
+            sage: SD = G.slice_decomposition(initial_vertex='x')
+            sage: SD.xslice_sequence('a')
+            [['a'], ['b', 'c', 'd', 'e', 'f'], ['g']]
+            sage: ('c', 'g') in SD.active_edges('a')
+            True
+            sage: ('a', 'c') in SD.active_edges('a')
+            True
+            sage: ('c', 'd') in SD.active_edges('a') # c and d in the same slice
+            False
+            sage: ('a', 'u') in SD.active_edges('a') # u not in x-slice of a
+            False
+
+        The set of active edges of every vertex is a partition of the edges::
+
+            sage: from itertools import chain
+            sage: E = list(chain(*(SD.active_edges(v) for v in G)))
+            sage: G.size() == len(E) == len(set(E)) \
+            ....:   and all(G.has_edge(u, w) for v in G for u, w in SD.active_edges(v))
+            True
+
+        TESTS:
+
+            sage: SD = graphs.PetersenGraph().slice_decomposition()
+            sage: SD.active_edges('Graham')
+            Traceback (most recent call last):
+            ...
+            LookupError: vertex (Graham) does not appear in the slice decomposition
+        """
+        if v not in self.sigma_inv:
+            raise LookupError(f"vertex ({v}) does not appear in the slice "
+                              "decomposition")
+        i = self.sigma_inv[v]
+        return self._xslice_active_edges(i)
+
+    def _slice(self, idx):
+        r"""
+        This method is for internal use only
+        """
```

Comment:
I think it's best to add a minimal indirect doctest even to internal methods.

Reply:
done in [5b263e4](https://github.com/sagemath/sage/pull/38299/commits/5b263e4e1a87d06180097bdd4751411e8be660c7)